src/ExperimentTable.js

```diff
@@ -103,33 +137,37 @@ class ExperimentTable extends React.Component {
   }
 
   render() {
-    const { searchQuery, searchedColumnIndex, selectedSearch, selectedKingdom, checkedRows } = this.state
+    const { searchQuery, searchedColumnIndex, selectedSearch, checkedRows } = this.state
+    const { selectedDropdownFilters, experimentTableFilters } = this.state
     const { orderedColumnIndex, ascendingOrder } = this.state
     const { entriesPerPage, currentPage } = this.state
     const { host, aaData, tableHeader, enableDownload, downloadTooltip } = this.props
 
     const displayedFields = tableHeader.map(header => header.dataParam)
 
-    const dataArray = selectedSearch.trim() ?
-
-      this.sort(aaData).filter(data => data &&
-        Object.keys(data).map(key => displayedFields.includes(key) ? data[key] : null)
-          .some(value => value && value.toString().toLowerCase().includes(selectedSearch))) :
-      this.filter(this.sort(aaData), tableHeader)
-        .filter(data => selectedKingdom ? data.kingdom === selectedKingdom : true)
+    //first we filter according to table headers
+    //then on those filtered experiments we further filter them in next step according to search box text
+    //and finally we filter the experiments filtered in previous two steps according to selected dropdowns
+    const filteredExperiments = this.filter(this.sort(aaData), tableHeader).filter(data =>
+      data && Object.keys(data).map(key =>
+        displayedFields.includes(key) ? data[key] : null).some(value =>
+        value && value.toString().toLowerCase().includes(selectedSearch))).filter(experiment =>
+      selectedDropdownFilters.every(filter =>
+        filter ? this.containsValue(experiment, filter.value) : true
+      ))
```

Comment:
How about this? Do you think it reads better?
```javascript
const filteredExperiments =
  this
    // First we filter according to table headers...
    .filter(this.sort(aaData), tableHeader)
    // ... then we further filter according to search box text...
    .filter(experiment =>
      experiment &&
      Object
        .keys(experiment)
        .map(key => displayedFields.includes(key) ? experiment[key] : null)
        .some(value => value && value.toString().toLowerCase().includes(selectedSearch.toLowerCase())))
    // ... and finally we filter according to selected dropdowns
    .filter(experiment =>
      selectedDropdownFilters.every(filter => filter ? this.containsValue(experiment, filter.value) : true))
```

This took me a while and a few tries. Don’t settle for the first thing that comes to mind. Try to be creative while consistent (for example, on your first `filter` block the argument is `data`, while in the second it’s `experiment`, when they’re the same elements). Shuffle things around, try different things.

Reply:
This definitely looks better. I have added a little touch of mine. Please review and let me know if it looks fine.